homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,282 @@
+"""Cover entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from dataclasses import dataclass, field
+import logging
+from typing import Any
+
+from motionblindsble.const import (
+    MotionBlindType,
+    MotionConnectionType,
+    MotionRunningType,
+)
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_CONNECTION,
+    CONF_BLIND_TYPE,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ICON_VERTICAL_BLIND,
+    MANUFACTURER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class MotionCoverEntityDescription(CoverEntityDescription):
+    """Entity description of a cover entity with default values."""
+
+    key: str = field(default=CoverDeviceClass.BLIND.value, init=False)
+    translation_key: str = field(default=CoverDeviceClass.BLIND.value, init=False)
+    device_class: CoverDeviceClass = field(default=CoverDeviceClass.SHADE, init=True)
+
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
+    MotionBlindType.ROLLER.value: MotionCoverEntityDescription(),
+    MotionBlindType.HONEYCOMB.value: MotionCoverEntityDescription(),
+    MotionBlindType.ROMAN.value: MotionCoverEntityDescription(),
+    MotionBlindType.VENETIAN.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.BLIND
+    ),
+    MotionBlindType.VENETIAN_TILT_ONLY.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.BLIND
+    ),
+    MotionBlindType.DOUBLE_ROLLER.value: MotionCoverEntityDescription(),
+    MotionBlindType.CURTAIN.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.CURTAIN
+    ),
+    MotionBlindType.VERTICAL.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up blind based on a config entry."""
+
+    blind_type = BLIND_TO_ENTITY_TYPE[entry.data[CONF_BLIND_TYPE]]
+    device = hass.data[DOMAIN][entry.entry_id]
+    blind = blind_type(device, entry)
+
+    async_add_entities([blind])
+
+
+class GenericBlind(CoverEntity):
+    """Representation of a blind."""
+
+    _attr_is_closed: bool | None = None
+    _attr_is_opening: bool | None = None
+    _attr_is_closing: bool | None = None
```

Comment:
```suggestion
```
those should be the defaults anyways

Reply:
> those should be the defaults anyways
Apparently for `_attr_is_closed` it is not? Currently I get this error: `AttributeError: 'PositionBlind' object has no attribute '__attr_is_closed'`, which is solved by adding the line again:

```
class GenericBlind(CoverEntity):
    """Representation of a blind."""

    _attr_is_closed: bool | None = None
    _attr_should_poll: bool = False
```

Reply:
The maintainers will tell you to at least remove the type hints (at least they told me), because those are defined in the base class. This leads to those attributes not being typed properly at least in my IDE, but I guess this is desired.

Reply:
I can remove the type hint of all the attributes defined in the parent classes except `_attr_is_closed`. If I remove the type hint I get this mypy error when I run pre-commit: `homeassistant/components/motionblinds_ble/cover.py:163: error: Incompatible types in assignment (expression has type "bool | None", variable has type "None")  [assignment]
Found 1 error in 1 file (checked 4 source files)`:

```
class GenericBlind(CoverEntity):
    """Representation of a blind."""

    _attr_is_closed = None
```

I can only solve this by including the type hint: `_attr_is_closed: bool | None = None`